### PR TITLE
Surgery tool indicators

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

This renames most surgery steps to tell which tool is needed, and also adds a feature where, whenever you have your mouse hovered over someone undergoing surgery, the tools that will work for the current step in your inventory will be highlighted.

The outlines will only render if you have a medhud or are a perfect/abductor surgeon.

## Why It's Good For The Game

Helps players who don't quite have surgery memorized figure out which tools to use.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

https://github.com/BeeStation/BeeStation-Hornet/assets/65794972/fa7d1c8e-a47c-40ab-9a41-10511ce71eaa

</details>

## Changelog
:cl:
tweak: Surgery steps on the operating computer now say which tool to use.
add: Added surgery tool helper outlines — whenever you have a medical HUD equipped, and you hover your mouse over someone undergoing surgery, all the tools in your inventory that can be used for the current surgery step will be highlighted with an outline, with the color of the outline representing the success rate of the tool.
/:cl:

### CODE REVIEW

**Review by Rukofamicom — APPROVED** [review]
This is a good QoL change

**Comment by llol111** [comment]
Is this when next to a surgery computer, or in general?
I have no issues with surgery computers doing some more handholding, but operating without one should rely on on momory (or wiki looking) imo.

**Comment by Rukofamicom** [comment]
> but operating without one should rely on on memory (or wiki looking) imo.

Disagree. It should rely on a *skills system of some sort* so players don't have to rely on a wiki to do basic things their character should have memorized and players acting OOC shouldn't even be able to do with much success.

**Comment by llol111** [comment]
If people dont want to look at the wiki they can learn the steps. I dont oppose a skillsystem exsiting if its well done, i'd like that actualy. But i dont think we should completely remove the need for *player* skill. 
I guess i'd be fine with something less overt, like how how construction/deconstruction gives you the next steps in flavor text (notably without highlighting the relevant tools directly)

**Comment by Absolucy** [comment]
@llol111 My two cents: as a medbay main, I see people just whacking every surgical tool against someone until one of them works _way_ too often.

... Also I keep messing up surgery whenever I'm an abductor, bc the alien organ replacement surgery swaps the hemostat and retractor for some reason and it screws with my muscle memory

**Comment by Absolucy** [comment]
based on a suggestion from the discord, I've made it so only people with medhuds will get the visual outlines now.

**Comment by llol111** [comment]
i like that change, it makes sense IC 👍 
fine with this then

**Comment by Absolucy** [comment]
@llol111 You okay with abductor scientists also being able to see the outline, since IIRC they're supposed to be "perfect surgeons", even though they don't have a medhud?

**Comment by llol111** [comment]
Maybe they could just be given a medhud implant, instead of snowflaking them?
idk, either way they're supposed to be hyper advanced greyliens, far above humanity... so i guess thats fine.
I just dont want regular people to have hints being given in such an incredibly gamey way.

File: code/modules/surgery/surgery.dm [hunk 1]
```diff
@@ -141,6 +147,86 @@
 	SSblackbox.record_feedback("tally", "surgeries_completed", 1, type)
 	qdel(src)
 
+/datum/surgery/proc/on_mouse_entered(datum/_source, mob/living/user)
+	SIGNAL_HANDLER
+	if(!istype(user) || user.zone_selected != location || !(HAS_TRAIT(user, TRAIT_MEDICAL_HUD) || HAS_TRAIT(user, TRAIT_SURGEON) || HAS_TRAIT(user, TRAIT_ABDUCTOR_SURGEON)) || (user in hovering_users))
```
Comment (Sarchutar) [inline code/modules/surgery/surgery.dm]: If i remember correctly, abductors get the surgeon trait for their mind and not the mob, so should be `user.mind` instead
or something like this if one wants to have both
`if(HAS_TRAIT(user, TRAIT_ABDUCTOR_SURGEON) || (user.mind && HAS_TRAIT(user.mind, TRAIT_ABDUCTOR_SURGEON)))`

**Comment by PowerfulBacon** [comment]
Perhaps there are better approaches than making it on mouse enter since you will only have your mouse over them after you have picked up the tool, meaning that when you move your mouse to get the item the outlines go away despite that being the time that you need them. Maybe it should be based on the surgery computer instead

**Comment by Absolucy** [comment]
Will eventually redo this in a better way.

**Comment by Tsar-Salat** [comment]
> Will eventually redo this in a better way. The dream is _not_ dead.

Just do screentips lol

**Comment by Absolucy** [comment]
screentips?

**Pull request closed without merging**